### PR TITLE
Pull offset check into updateStorageValue

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/YulUtilFunctions.cpp [hunk 1]
```diff
@@ -1274,7 +1274,6 @@ string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
 {
 	solAssert(_type.location() == DataLocation::Storage, "");
 	solAssert(_type.isDynamicallySized(), "");
-	solUnimplementedAssert(_type.baseType()->storageBytes() <= 32, "Base type is not yet implemented.");
```
Comment (chriseth) [inline libsolidity/codegen/YulUtilFunctions.cpp:1277]: I don't know what check this was supposed to be... ;)

File: test/libsolidity/semanticTests/array/copying/array_copy_including_array.sol [hunk 2]
```diff
@@ -33,6 +33,8 @@ contract c {
         return (small.length, large.length);
     }
 }
+// ====
+// compileViaYul: also
```
Comment (mijovic) [inline test/libsolidity/semanticTests/array/copying/array_copy_including_array.sol:37]: This test works locally for you?

Comment (chriseth) [reply test/libsolidity/semanticTests/array/copying/array_copy_including_array.sol:37]: I guess it worked when it was still rebased on top of another PR?

**Comment by chriseth** [comment]
Rebased. Hopefully the tests work now.

File: libsolidity/codegen/YulUtilFunctions.cpp (hunk 1 above)
Comment (mijovic) [reply libsolidity/codegen/YulUtilFunctions.cpp:1277]: I don't know as well, very helpful message in assert as well :)

File: test/libsolidity/semanticTests/array/copying/array_copy_including_array.sol (hunk 2 above)
Comment (mijovic) [reply test/libsolidity/semanticTests/array/copying/array_copy_including_array.sol:37]: Probably, looks like it is solved now

**Review by mijovic — APPROVED** [review]
Looks good

**Pull request merged**